src/IB/IBFEMethod.cpp

```diff
@@ -1011,12 +1025,47 @@ IBFEMethod::spreadForce(const int f_data_idx,
             const IntVector<NDIM> gcw =
                 level->getPatchDescriptor()->getPatchDataFactory(f_scratch_data_idx)->getGhostCellWidth();
 
+            // TODO - SAMRAIGhostDataAccumulator has not been tested with multiple levels
             d_ghost_data_accumulator.reset(new SAMRAIGhostDataAccumulator(
-                hierarchy, f_var, gcw, d_hierarchy->getFinestLevelNumber(), d_hierarchy->getFinestLevelNumber()));
+                hierarchy, f_var, gcw, getCoarsestPatchLevelNumber(), getFinestPatchLevelNumber()));
         }
         d_ghost_data_accumulator->accumulateGhostData(f_scratch_data_idx);
     }
 
+    // Prolong forces spread onto coarser levels onto finer levels.
+    if (d_use_scratch_hierarchy)
+    {
+        // not yet implemented
+        assertStructureOnFinestLevel();
+    }
+    else
+    {
+        // TODO - we should save these schedules in the same manner that we save
+        // the schedules for the scratch hierarchy.
+        for (int ln = getCoarsestPatchLevelNumber(); ln < getFinestPatchLevelNumber(); ++ln)
+        {
+            Pointer<RefineAlgorithm<NDIM> > refine_algorithm = new RefineAlgorithm<NDIM>();
+            Pointer<CartesianGridGeometry<NDIM> > geometry = d_hierarchy->getGridGeometry();
+            // TODO: this is probably the wrong refinement operator
+            Pointer<RefineOperator<NDIM> > refine_op =
+                geometry->lookupRefineOperator(f_var, "CONSERVATIVE_LINEAR_REFINE");
+            TBOX_ASSERT(refine_op);
+            refine_algorithm->registerRefine(
+                f_prolong_scratch_data_idx, f_scratch_data_idx, f_prolong_scratch_scratch_data_idx, refine_op);
+            Pointer<PatchLevel<NDIM> > coarse_level = d_hierarchy->getPatchLevel(ln);
+            Pointer<PatchLevel<NDIM> > fine_level = d_hierarchy->getPatchLevel(ln + 1);
+            // We can ignore the fourth argument since we don't need to deal with
+            // forces outside the physical domain (this was handled above by f_phys_bdry_op)
```

Comment:
Maybe --- maybe not --- depending on the refine operator you may need boundary values close to the boundary. This is one reason to prefer to use RT0 here.

Reply:
Good point. I'll get back to this later once the prolongation is working.

Reply:
I thought about this some more (and I had to reread some other comments I left in IBFEMethod after #896): since we spread into ghost regions that f_phys_bdry_op would in turn accumulate into another ghost region we need to handle this part prior to the global 'sum forces spread into ghost regions into the correct patch interiors' step. Hence - at this point in the routine we have already accumulated all forces so we should deliberately ignore these values.